stack/eslint-plugin/src/no-new-null.ts

```diff
@@ -54,7 +55,7 @@ const noNewNullRule: TSESLint.RuleModule<MessageIds, Options> = {
       switch ((node as TSESTree.Node).type) {
         case AST_NODE_TYPES.MethodDefinition:
           return true;
-        case AST_NODE_TYPES.ClassProperty:
+        case AST_NODE_TYPES.PropertyDefinition:
```

Comment:
This doesn't break compatibility with 6 or 7, right?

Reply:
No, it shouldn't be. This type is sourced from the @typescript-eslint/ast-spec package, which in-turn is used by the parser. The type that we get back would come from `@typescript-eslint/typescript-estree`, which also uses that package as long as we keep them updated to the same version (which we do in this PR).

Here's the breaking change that necessitated this update: https://github.com/typescript-eslint/typescript-eslint/pull/3806